From f69a7bf85600b1a01d952d9c28200363b87d94cb Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 10 Jun 2022 16:55:46 +0200
Subject: [PATCH 0261/1695] dt-bindings: introduce RISAB device tree bindings

This change introduces RISABPROT macro which describes the RIF
configuration for a RISAB, and various defines.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ec2ddbfe4799a7d054969d2be51f002c57b93748 dt-bindings: introduce RISAB device tree bindings]
Change-Id: I6a8ec34a59863657d9007fa5e5960f0b15274f54
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
---
 .../include/dt-bindings/soc/stm32mp25-risab.h | 36 +++++++++++++++++++
 1 file changed, 36 insertions(+)
 create mode 100644 core/include/dt-bindings/soc/stm32mp25-risab.h

diff --git a/core/include/dt-bindings/soc/stm32mp25-risab.h b/core/include/dt-bindings/soc/stm32mp25-risab.h
new file mode 100644
index 000000000..a615bb259
--- /dev/null
+++ b/core/include/dt-bindings/soc/stm32mp25-risab.h
@@ -0,0 +1,36 @@
+/* SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause) */
+/*
+ * Copyright (C) 2022, STMicroelectronics - All Rights Reserved
+ */
+
+#ifndef _DT_BINDINGS_STM32MP25_RISAB_H
+#define _DT_BINDINGS_STM32MP25_RISAB_H
+
+/* RISAB control modes */
+#define RIF_DDCID_DIS				0x0
+#define RIF_DDCID_EN				0x1
+
+#define RISAB_READ_LIST_SHIFT			8
+#define RISAB_WRITE_LIST_SHIFT			16
+#define RISAB_CFEN_SHIFT			24
+#define RISAB_DPRIV_SHIFT			25
+#define RISAB_SEC_SHIFT				26
+#define RISAB_DCCID_SHIFT			27
+#define RISAB_DCEN_SHIFT			31
+
+#define RISABPROT(delegate_en, delegate_cid, sec, default_priv, \
+		  enabled, cid_read_list, cid_write_list, cid_priv_list) \
+	(((delegate_en) << RISAB_DCEN_SHIFT) | \
+	 ((delegate_cid) << RISAB_DCCID_SHIFT) | \
+	 ((sec) << RISAB_SEC_SHIFT) | ((default_priv) << RISAB_DPRIV_SHIFT) | \
+	 ((enabled) << RISAB_CFEN_SHIFT) | \
+	 ((cid_write_list) << RISAB_WRITE_LIST_SHIFT) | \
+	 ((cid_read_list) << RISAB_READ_LIST_SHIFT) | (cid_priv_list))
+
+/* RISABPROT macro masks */
+#define RISAB_PLIST_MASK			GENMASK_32(7, 0)
+#define RISAB_RLIST_MASK			GENMASK_32(15, 8)
+#define RISAB_WLIST_MASK			GENMASK_32(23, 16)
+#define RISAB_DCCID_MASK			GENMASK_32(30, 27)
+
+#endif /* _DT_BINDINGS_STM32MP25_RISAB_H */
-- 
2.47.2

